src/tritonserver.cc

```diff
@@ -2731,12 +2731,17 @@ TRITONSERVER_ServerInferAsync(
 TRITONSERVER_Error*
 TRITONSERVER_MetricFamilyNew(
     TRITONSERVER_MetricFamily** family, TRITONSERVER_MetricKind kind,
-    const char* name, const char* description)
+    const char* name, const char* description, void* registry_ptr)
 {
 #ifdef TRITON_ENABLE_METRICS
-  const auto& registry = tc::Metrics::GetRegistry();
+  const auto& registry =
+      reinterpret_cast<std::shared_ptr<prometheus::Registry>*>(registry_ptr);
+  // TODO: Remove this after review
+  // TODO: Verify passing registry_ptr from API caller is OK.
+  //       Calling GetRegistry() here did not add the new metrics to registry.
+  // const auto& registry = tc::Metrics::GetRegistry();
```

Comment:
Need another set of eyes on this. 

Originally I retrieved the Registry object in the API implementation here:
```
const auto& registry = tc::Metrics::GetRegistry();
*family = reinterpret_cast<TRITONSERVER_MetricFamily*>(
      new tc::MetricFamily(kind, name, description, registry))
```

But the new metrics did not show up when collecting from the registry.

However, by passing the registry object from the API caller (unit test in this case), the new metrics were correctly added to the registry when calling `triton::core::Metrics::SerializedMetrics();`

Reply:
Resolved by using C APIs to get metrics like `TRITONSERVER_ServerMetrics` and `TRITONSERVER_MetricsFormatted` instead of `tc::Metrics::SerializedMetrics()` in the unit test